torch_xla/csrc/aten_xla_type.cpp

```diff
@@ -1279,9 +1279,8 @@ at::Tensor XLANativeFunctions::hardtanh_backward(const at::Tensor& grad_output,
       max_val));
 }
 
-at::Tensor XLANativeFunctions::index(
-    const at::Tensor& self,
-    const c10::List<c10::optional<at::Tensor>>& indices) {
+at::Tensor XLANativeFunctions::index(const at::Tensor& self,
+                                     const at::IOptTensorListRef& indices) {
   XLA_FN_COUNTER("xla::");
   bool indices_on_cpu_or_xla =
       std::all_of(indices.begin(), indices.end(),
```

Comment:
The iterator of `IOptTensorListRef` is an `OptionalTensorRef`, and not a `optional<at::Tensor>`.
So, instead of `[=](const c10::optional<at::Tensor>& opt)`, it should be `[=](OptionalTensorRef opt)`. Note that, in this context, the iterated `OptionalTensorRef` is a temporary, so you can't get a reference, here.

Reply:
Hmmm... Maybe its lifetime is extended, here...